Makefile

```diff
@@ -98,6 +98,10 @@ OPT += -momit-leaf-frame-pointer
 endif
 endif
 
+ifeq (,$(shell $(CXX) -fsyntax-only -faligned-new -xc++ /dev/null 2>&1))
+CXXFLAGS += -faligned-new -DHAVE_ALIGNED_NEW
```

Comment:
@grooverdan  We start to get the following compile error with clang_analyzer (clang-6.0):
```
cache/lru_cache.cc:118: error: undefined reference to 'operator delete(void*, std::align_val_t)'
monitoring/statistics.cc:28: error: undefined reference to 'operator delete(void*, std::align_val_t)'
```
The destructors are now calling c++17 delete operator, and seems to related to the faligned-new flag. Mind take a look? Thanks.

Reply:
sure. Added #4075 to cover compile testing of clang-6,7,gcc-8

Reply:
what was the compile that triggered it?

Reply:
you can try 
```
USE_CLANG=1 OPT=-g make -j1 analyze 
```